client/components/place/places-location/locationPlaces.component.js

```diff
@@ -1,30 +1,50 @@
-
-locationplacesCtrl.$inject = ['mapFactory', '$scope'];
-function locationplacesCtrl(mapFactory, $scope) {
+function locationplacesCtrl(mapFactory, $scope, $rootScope, placesOnMap, constants, Place, $timeout) {
   var ctrl = $scope;
-  ctrl.places = this.places;
-  $scope.placesByLocation = ctrl.places;
+  var latitude;
+  var longitude;
+  Place.getList({sort: '-rate', limit: 10}).then(function(popularPlaces) {
+    ctrl.popularPlaces = popularPlaces;
+    $scope.placesByLocation = popularPlaces;
+    $scope.placesByLocation = _.shuffle($scope.placesByLocation);
+  });
+
   $scope.map = mapFactory.map;
   $scope.map.on('moveend', setLocationPlaces);
+
   function setLocationPlaces() {
     $scope.center = $scope.map.getCenter();
     var bounds = $scope.map.getBounds();
     var min = bounds.getSouthWest().wrap();
     var max = bounds.getNorthEast().wrap();
     $scope.placesByLocation = [];
-    for (i = 0; i < ctrl.places.length; i++) {
-      if (ctrl.places[i].latitude > min.lat && ctrl.places[i].longitude > min.lng && ctrl.places[i].latitude < max.lat && ctrl.places[i].longitude < max.lng) {
-        $scope.placesByLocation.push(ctrl.places[i]);
-        $scope.$apply();
+    ctrl.places = placesOnMap.getPlaceArr();
+    var placesArr = ctrl.places;
+
+    $scope.placesByLocation = placesArr.filter(function(place) {
+        latitude = place.location.coordinates[1];
+        longitude = place.location.coordinates[0];
+        return (latitude > min.lat && longitude > min.lng
+          && latitude < max.lat && longitude < max.lng) && place.photos[0];
       }
-    }
+    );
+    $scope.placesByLocation = _.sortBy($scope.placesByLocation, ['rate']);
```

Comment:
Yes, it's no nesseserity in sorting inside filter, sorting (_.sortBy) is for taking only 5 rating places ( it's not usefull right now but if we'll do preloading or smth else, it would be usefull). I think, it no good getting in scope very big array